metricflow/sql/optimizer/rewriting_sub_query_reducer.py

```diff
@@ -196,7 +199,7 @@ def _is_simple_source(node: SqlSelectStatementNode) -> bool:
             if select_column.expr.lineage.contains_aggregate_exprs:
                 return False
         return (
-            len(node.parent_nodes) <= 1
+            len(node.join_descs) == 0
```

Comment:
This looks like a behavior change - were there changes to the snapshots from this change alone?

Reply:
There were no snapshot changes. This is not a behavior change as without CTEs, the number of parent nodes is 1 + the number of joins.